packages/flask/src/UICarouselView/CarouselView/CarouselContainer/index.web.tsx

```diff
@@ -0,0 +1,189 @@
+    shouldPageMoveOnPress = true,
+    onPageIndexChange,
+}: Props) => {
+    const [layout, setLayout] = React.useState({ width: 0, height: 0 });
+
+    const carouselOffset = useSharedValue(0);
+    const itemOffset = useSharedValue(0);
+
+    const layoutRef = React.useRef(layout);
+    const onPageChangedRef = React.useRef(onPageIndexChange);
+    const currentIndexRef = React.useRef(0);
+    const maxWidthTranslate = React.useMemo(() => layout.width * pages.length, [layout, pages]);
+
+    const [isMoving, setIsMoving] = React.useState(false);
+
+    const containerAnimatedStyle = useAnimatedStyle(() => {
+        const transform = [
+            {
+                translateX: interpolate(
+                    carouselOffset.value,
+                    [-maxWidthTranslate, 0],
+                    [-maxWidthTranslate, 0],
+                ),
+            },
+        ];
+
+        return { transform };
+    });
+
+    const jumpToIndex = React.useCallback(
+        (index: number) => {
+            const offset = -index * layout.width;
+            carouselOffset.value = withTiming(offset, { duration });
+            itemOffset.value = withSequence(
+                withTiming(1, { duration: duration / 2 }),
+                withTiming(0, { duration }),
+            );
+            setIsMoving(false);
+            onPageChangedRef.current && onPageChangedRef.current(index);
+        },
+        [carouselOffset, itemOffset, layout.width],
+    );
+
+    const jumpToNext = React.useCallback(() => {
+        if (!isMoving) {
+            const nextIndex = (currentIndexRef.current + 1) % pages.length;
+            jumpToIndex(nextIndex);
+        }
+    }, [jumpToIndex, isMoving, pages]);
+
+    React.useEffect(() => {
+        layoutRef.current = layout;
+        const offset = -currentIndexRef.current * layoutRef.current.width;
+        if (onPageIndexChange) {
+            onPageChangedRef.current = onPageIndexChange;
+        }
+        carouselOffset.value = offset;
+    }, [layoutRef, carouselOffset, onPageIndexChange, layout]);
+
+    React.useEffect(() => {
+        if (layoutRef.current.width && currentIndexRef.current !== initialIndex) {
+            currentIndexRef.current = initialIndex;
+            jumpToIndex(initialIndex);
+        }
+    }, [jumpToIndex, layoutRef.current.width, initialIndex]);
+
+    const handleLayout = React.useCallback((e: LayoutChangeEvent) => {
+        const { height, width } = e.nativeEvent.layout;
+        setLayout(prevLayout => {
+            if (prevLayout.width === width && prevLayout.height === height) {
+                return prevLayout;
+            }
+            return { height, width };
+        });
+    }, []);
+
+    return (
+        <View testID={testID} onLayout={handleLayout} style={[styles.pager]}>
+            <Animated.View
+                style={[
+                    styles.sheet,
+                    layout.width ? { width: maxWidthTranslate, ...containerAnimatedStyle } : null,
```

Comment:
```suggestion
                    layout.width ? [{ width: maxWidthTranslate }, containerAnimatedStyle ] : null,
```

Reply:
@savelichalex No, it didn't work that way